tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -168,5 +188,20 @@ private static void WriteRequestPayload(WebRequest request, string data)
             dataStream.Write(byteArray, 0, byteArray.Length);
             dataStream.Close();
         }
+
+        private static void WriteRequestHeaders(WebRequest request, IDictionary<string, string> context)
+        {
+            foreach (KeyValuePair<string, string> kv in context)
+            {
+                request.Headers.Add(kv.Key, kv.Value);
+            }
+        }
+
+        private static object GetTraceContext(object obj)
+        {
+            // Datadog duck typing library
+            var proxyInstance = obj.DuckAs<ILambdaContext>();
+            return proxyInstance.ClientContext.Custom;
```

Comment:
`DuckAs<T>` returns `null` if the operation fails. I think it's more efficient to _not_ check the type above (`if (eventOrContext.GetType().ToString().Contains("LambdaContext")`), and instead let duck typing handle that and return `null` when `obj` is not a "LambdaContext".

This `?` operator avoids a null reference exception if `proxyInstance` is `null`. Instead of throwing an exception, `GetTraceContext()` will return `null`.
```suggestion
            return proxyInstance?.ClientContext.Custom;
```